src/FM.LiveSwitch.Mux.Standard/Connection.cs

```diff
@@ -246,14 +246,12 @@ public async Task<bool> ProcessLogEntry(LogEntry logEntry, MuxOptions options)
                     }
                 }
 
-                var videoDelay = logEntry.Data?.VideoDelay ?? 0D;
-                if (videoDelay != 0 &&
-                    ActiveRecording.VideoStartTimestamp.HasValue &&
-                    ActiveRecording.VideoStopTimestamp.HasValue)
-                {
-                    ActiveRecording.VideoStartTimestamp = ActiveRecording.VideoStartTimestamp.Value.AddSeconds(videoDelay);
-                    ActiveRecording.VideoStopTimestamp = ActiveRecording.VideoStopTimestamp.Value.AddSeconds(videoDelay);
-                }
+                // No longer adding the video delay to the video timestamp
+                // The video/audio start timestamps are set from when the server processed the relevant
+                // frame while the delay is calculated from the rtp/rtcp packet timestamps.
+                // This means there is no relationship between the start timestamps in the event
+                // log and the delay. The event timestamps should already account for the necessary delay
+                // between the streams when muxing.
```

Comment:
Nice deductions here!

Today, the audio/video start timestamps are the _server-side_ wall-clock times measured when the first packet is received on each track. The reason this solution sorta works is that, in general, this should be close to the true offset, especially if the audio and video tracks use the same network path from the client to the server.

It will not always be the case, though, and I think there is a better solution... but it requires additional information to be written by the Media Server to the log file.

We actually want the audio/video start timestamps to be a reflection of the _client-side_ wall-clock times measured when the first frame is captured on each track. I think we have enough information in the Media Server to incorporate this into the value of `VideoDelay` such that it can be reliably applied to the start timestamps.

I am hesitant to merge this in until we investigate that path.

Reply:
When I was looking at the history of how the timestamps were set, the media server originally marked the start timestamps for each stream using the client wallclock (from the incoming packets) but it was eventually changed to use the server's utc.Now (though the delay wasn't updated at the same time - hence this issue). My assumption was that this was to aid synchronizing the streams from different clients (whose clocks are likely not aligned).

Assuming the above is correct, if we keep the server timestamps in the log, but add the client timestamps for the streams back in, it may be possible to have ls mux use the client timestamps+delay to mux the audio/video for each individual stream, and then the server timestamps to mux each of the client streams together.

As an aside, I was wondering if it would be possible to treat recording as just another client (from the perspective of the media server at least) which outputs to a file rather than a display - as the client's do not seem to have nearly as many issues around synchronization and we could avoid all these separate paths for recordings. Not sure if that's feasible though, and it would be a large architectural change.

Reply:
Reverted the change to skip adding the video delay in light of the corrected calculation on the media server